examples/02_decoding/plot_oasis_vbm.py

```diff
@@ -39,132 +45,139 @@
+
+# Preprocess the mask: remove features with too low between-subject variance
+from sklearn.feature_selection import VarianceThreshold
 from nilearn.input_data import NiftiMasker
-
-n_subjects = 100  # more subjects requires more memory
-
-############################################################################
-# Load Oasis dataset
-# -------------------
-oasis_dataset = datasets.fetch_oasis_vbm(n_subjects=n_subjects)
-gray_matter_map_filenames = oasis_dataset.gray_matter_maps
-age = oasis_dataset.ext_vars['age'].astype(float)
-
-# print basic information on the dataset
-print('First gray-matter anatomy image (3D) is located at: %s' %
-      oasis_dataset.gray_matter_maps[0])  # 3D data
-print('First white-matter anatomy image (3D) is located at: %s' %
-      oasis_dataset.white_matter_maps[0])  # 3D data
-
-#############################################################################
-# Preprocess data
-# ----------------
-nifti_masker = NiftiMasker(
-    standardize=False,
-    smoothing_fwhm=2,
-    memory='nilearn_cache')  # cache options
-gm_maps_masked = nifti_masker.fit_transform(gray_matter_map_filenames)
-n_samples, n_features = gm_maps_masked.shape
-print("%d samples, %d features" % (n_subjects, n_features))
-
-############################################################################
-# Prediction pipeline with ANOVA and SVR
-# ---------------------------------------
-print("ANOVA + SVR")
-# Define the prediction function to be used.
-# Here we use a Support Vector Classification, with a linear kernel
-from sklearn.svm import SVR
-svr = SVR(kernel='linear')
-
-# Dimension reduction
-from sklearn.feature_selection import VarianceThreshold, SelectKBest, \
-        f_regression
-
-# Remove features with too low between-subject variance
 variance_threshold = VarianceThreshold(threshold=.01)
+nifti_masker = NiftiMasker(standardize=False, smoothing_fwhm=2,
+                           memory='nilearn_cache')
+
+gm_maps_masked = nifti_masker.fit_transform(gm_imgs_train)
+
+# Fit the masker again to remove features with too low between-subject variance
+gm_maps_thresholded = variance_threshold.fit_transform(gm_maps_masked)
+gm_maps_masked = variance_threshold.inverse_transform(gm_maps_thresholded)
+
+mask = nifti_masker.inverse_transform(variance_threshold.get_support())
+###############################################################################
+# Training the decoder
+######################################################################
+
+# To save time (because these are anat images with many voxels), we include
+# only the 5-percent voxels most correlated with the age variable to fit.
+# Also, we set memory_level=2 so that more of the intermediate computations
+# are cached. Also, you may pass and n_jobs=<some_high_value> to the
+# DecoderRegressor class, to take advantage of a multi-core system.
+from nilearn.decoding import DecoderRegressor
+decoder = DecoderRegressor(estimator='svr', mask=mask,
```

Comment:
In this example there is a `variance_threshold` object to remove all features with too-low variance after masking the images, so I think it is more beneficial to keep the `mask` image as input to `Decoder` instead of masker object.

Reply:
I think that  using thresholds is a different discussion than that of accepting `NiftiMasker` or `mask_img`

Reply:
>In this example there is a variance_threshold object to remove all features with too-low variance after masking the images, so I think it is more beneficial to keep the mask image as input to Decoder instead of masker object.

Yeah makes sense. I think this NiftiMasker mask computation is to select relevant features using VarianceThreshold on 2D.

Reply:
Should we add a variance threshold to the decoder? I see that it could be useful in many cases.

Reply:
What's the consensus on this?

Reply:
I'd keep now masking heuristics for future work I guess.